hera_pspec/uvwindow.py

```diff
@@ -75,152 +78,153 @@ def __init__(self, pol, spw_range=None, ftfile=None,
+        ftfile : str
+            Path to file constraining to the Fourier transform of the 
+            beam on the sky plane (Eq. 10 in Memo). The input is the 
+            root name of the file to use, including the polarisation
+            Ex : path/to/file/ft_beam_HERA_dipole_pI.hdf5
+            File must be h5 format.
 
         Returns
         ----------
-        mapsize : float
-            Size of the flat map the beam was projected onto (in deg).
-        ft_beam : array_like
-            Real part of the Fourier transform of the beam taken on the sky
-            plane (flat-sky approximation), for each of the frequencies
-            along the spectral window considered.
-            Has dimensions (Nfreqs,N,N).
+        bandwidth : array of floats
+            List of frequencies covered by the instrument, in Hz.
         """
-        assert self.ft_file is not None,\
-            "Can only be used if self.ft_file is defined."
+        if not os.path.isfile(ftfile):
+             raise ValueError("Cannot find FT beam file: {}.".format(ftfile))
 
-        filename = '{}_{}.hdf5'.format(self.ft_file, self.pol)
-        assert os.path.isfile(filename), \
-            "Cannot find FT beam file: {}.".format(filename)
-        with h5py.File(filename, "r") as f:
-            mapsize = f['mapsize'][0]
-            ft_beam = np.array(f['FT_beam'][self.spw_range[0]:self.spw_range[1], :, :])
+        with h5py.File(ftfile, "r") as f:
+            bandwidth = np.array(f['freq'][...])
 
-        return mapsize, ft_beam
+        assert bandwidth.size > 1, "Error reading file, empty bandwidth."
+
+        return bandwidth
 
     def update_spw(self, spw_range):
         """
-        Function to extract spectrl window from FTBeam defined on whole bandwidth.
+        Function to extract spectral window from FTBeam defined on whole bandwidth.
 
         Extract a section of the previously computed Fourier
         transform of the beam.
 
         Parameters
         ----------
-        spw_range : tuple or array
+        spw_range : tuple or 2-element array of ints.
             In (start_chan, end_chan). Must be between 0 and 1024 (HERA
             bandwidth).
         """
         # checks on inputs
-        spw_range = np.array(spw_range, dtype=int)
-        assert spw_range.size == 2, "spw_range must be fed as a tuple of "\
-                                    "frequency indices."
-        assert spw_range[1]-spw_range[0] > 0, \
-            "Require non-zero spectral range."
-        assert min(spw_range) >= 0 and max(spw_range) < self.freq_array.size,\
-            "spw_range must be integers within the given bandwith."
-        assert self.ft_beam is not None,\
-            "The FT of the beam needs to have already been read or computed"
+        assert check_spw_range(spw_range, self.freq_array), \
+            "Wrong spw range format, see dosctring."
+        assert self.ft_beam is not None, "No ft_beam attribute."
```

Comment:
I think this could never fail, right? Since ft_beam is always part of the instance now.

Reply:
I think it could. Imagine you've initialized a `FTBeam` object on a reduced bandwidth and try to reduce it even more.

I know you don't like the `update_something` functions but this one is necessary inside `from_uvpspec` because of the limited spectral info available inside the `UVPSpec` objects

Reply:
Oh I misunderstood your comment and thought you were referring to `assert check_spw_range(spw_range, self.freq_array)`. You are right, there is no reason for `self.ft_beam` to be None in the new version of the `FTBeam` class.